lib/librato.js

```diff
@@ -267,23 +314,35 @@ var flush_stats = function librato_flush(ts, metrics)
     if (brokenMetrics[measure.name]) {
       return;
     }
-
-    if (mType == 'counter') {
-      counters.push(measure);
+    
+    if (multidimensional) {
+      measurements.push(measure);
+      
+      // Post MD measurement
+      if (measurements.length >= maxBatchSize) {
+        post_measurements(measureTime, measurements);
```

Comment:
I don't see a matching call to this at the end of flush() for any remaining measurements left in the payload.